tests/controllers/DMSCartAbstractControllerTest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * Class DMSDocumentCartControllerTest contains all the tests for {@link DMSDocumentCartController}
+ */
+class DMSCartAbstractControllerTest extends FunctionalTest
+{
+    protected static $fixture_file = 'dms-cart/tests/DMSDocumentCartTest.yml';
+
+    /**
+     * @var DMSCartAbstractController
+     */
+    protected $controller;
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->controller = DMSCartAbstractController::create();
+    }
+
+    /**
+     * Ensure the link is "friendly", not a class name
+     */
+    public function testLink()
+    {
+        $this->controller = DMSCheckoutController::create();
+        $this->assertSame('checkout', $this->controller->Link());
+        $this->assertSame('checkout/complete', $this->controller->Link('complete'));
```

Comment:
This behaviour belongs in the checkout controller test class - this test class should test the logic in the abstract controller